pkg/contexts/ocm/accessmethods/plugin/plugin.go

```diff
@@ -74,6 +76,47 @@ func (p *PluginHandler) AccessMethod(spec *AccessSpec, cv cpi.ComponentVersionAc
 	return newMethod(p, spec, cv.GetContext(), info, creddata), nil
 }
 
+func (p *PluginHandler) GetInexpensiveContentVersionIdentity(spec *AccessSpec, cv cpi.ComponentVersionAccess) string {
+	mspec := p.GetAccessMethodDescriptor(spec.GetKind(), spec.GetVersion())
+	if mspec == nil {
+		return "unknown type " + spec.GetType()
+	}
+
+	if !mspec.SupportContentIdentity {
+		return ""
+	}
+
+	info, err := p.Info(spec)
+	if err != nil {
+		return ""
+	}
+
+	var creds credentials.Credentials
+	if len(info.ConsumerId) > 0 {
+		creds, err = credentials.CredentialsForConsumer(cv.GetContext(), info.ConsumerId, hostpath.IdentityMatcher(info.ConsumerId.Type()))
+		if err != nil {
+			return ""
+		}
+	}
+
+	var creddata json.RawMessage
+	if creds != nil {
+		creddata, err = json.Marshal(creds)
+		if err != nil {
+			return "cannot marshal creds"
+		}
+	}
+	specdata, err := spec.GetRaw()
+	if err != nil {
+		return ""
+	}
```

Comment:
All this should be the same for the Reader implementation, Can we dedup this coding?